### ISSUE TITLE
Fix GitHub Actions to automatically deploy changes to Pages upon merge

### ISSUE THREAD

**Issue by andrablaj** [medic/cht-tech-radar-contributors#3]
Currently, the GitHub Action for deploying the latest merged changes to GitHub Pages is broken.

<img width="1006" alt="Screenshot 2023-09-29 at 13 04 29" src="https://github.com/medic/cht-tech-radar-contributors/assets/24651706/30fb068b-267f-40a8-a06c-c2703f698375">

See errors [here](https://github.com/medic/cht-tech-radar-contributors/actions/runs/6328202725).

The same fix should be applied to [medic/cht-tech-radar-implementers](https://github.com/medic/cht-tech-radar-implementers)

**Comment by andrablaj** [comment]
Possible solution [here](https://www.kevinhooke.com/2022/11/10/gp-pages-deploy-fails-when-run-from-github-action-author-identity-unknown/).

**Comment by andrablaj** [comment]
@garethbowen This is not urgent, but I would love to look into this GH Actions error. The solution provided [here](https://www.kevinhooke.com/2022/11/10/gp-pages-deploy-fails-when-run-from-github-action-author-identity-unknown/), but I would love to hear your thoughts first; maybe you've encountered this problem in the past.

**Comment by garethbowen** [comment]
In my experience GH Pages automatically deploys with an action that's created by GH - you don't need to call it yourself. For example, [this action](https://github.com/medic/cht-tech-radar-contributors/actions/runs/7170759934/job/19524246010) appears to be also doing the deploy.

However because we want to do a compile step before deploying the default action won't work - we may need to delete this action?

Other than that I have no experience. I would try to copy with the cht-docs build does here: https://github.com/medic/cht-docs/blob/main/.github/workflows/ci.yml#L39-L48

**Comment by andrablaj** [comment]
>[this action](https://github.com/medic/cht-tech-radar-contributors/actions/runs/7170759934/job/19524246010) appears to be also doing the deploy.

That action is triggered in my local environment by following [these steps](https://github.com/medic/cht-tech-radar-contributors?tab=readme-ov-file#deployment-to-github-pages), as a workaround of the CI being broken.  

> I would try to copy with the cht-docs build does here: https://github.com/medic/cht-docs/blob/main/.github/workflows/ci.yml#L39-L48

Good lead! I will dig back into this when I get some focus time.

**Comment by andrablaj** [comment]
@garethbowen, per [your example](https://github.com/medic/cht-docs/blob/main/.github/workflows/ci.yml#L39-L48),can you I setup the `DEPOLY_TO_SITE` secret to this repo - as I don't have access to cht-docs one? (also, there's a typo in the name of the secret)

**Comment by andrablaj** [comment]
TL;DR I changed the CI with the same code as [cht-docs uses](https://github.com/medic/cht-docs/blob/main/.github/workflows/ci.yml#L39-L48), pushed to `main` branch on this repo and broke our docs site. 🤦‍♀️  [More details in this ticket](https://github.com/medic/cht-docs/issues/1341). I will get back to this with a fresh mind, to understand how to push the changes to GitHub Pages without completely replacing the docs site with Tech Radar content.

**Comment by garethbowen** [comment]
@andrablaj I suspect it's something in the configuration of the GH Pages for the repo. There are two other repos that publish to a subdir of the docs site, eg: https://github.com/medic/build-history -> https://docs.communityhealthtoolkit.org/build-history/

Let me know if/when you're blocked on me.

**Comment by andrablaj** [comment]
Adding a `GH_SECRET` by following the steps outlined in [this ticket](https://github.com/medic/medic-infrastructure/issues/1000#issue-2365244153), but with access to `cht-tech-radar-contributors` & `cht-tech-radar-implementers` solved the problem.
The deployment to GH Pages works correctly now via GH Actions.
